Directory.Build.props

```diff
@@ -1,24 +1,31 @@
 <Project>
   <PropertyGroup>
-    <TreatWarningsAsErrors>False</TreatWarningsAsErrors>
 
     <LangVersion>7.2</LangVersion>
     <!-- This is strong naming, not signing-->
     <SignAssembly>true</SignAssembly>
     <!-- The MSAL.snk has both private and public keys -->
     <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)/build/MSAL.snk</AssemblyOriginatorKeyFile>
     <DelaySign>false</DelaySign>
-    <CodeAnalysisRuleSet>$(MSBuildThisFileDirectory)/build/SolutionWideAnalyzerConfig.ruleset</CodeAnalysisRuleSet>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
     <IsPackable>false</IsPackable>
+    <!-- Configure SDL rulesets to use. https://www.1eswiki.com/wiki/Microsoft.Internal.Analyzers.Rulesets -->
+    <MicrosoftInternalSdlRuleset>Recommended</MicrosoftInternalSdlRuleset>
+    <MicrosoftInternalSdlRulesetAction>Error</MicrosoftInternalSdlRulesetAction>
+    <!--<MicrosoftInternalDisableNonSdlRules>false</MicrosoftInternalDisableNonSdlRules>-->
+    <AdditionalAnalyzerRuleSet>$(MSBuildThisFileDirectory)/build/SolutionWideAnalyzerConfig.ruleset</AdditionalAnalyzerRuleSet>
   </PropertyGroup>
   <ItemGroup>
     <!-- PrivateAssets="All" is on these packages to ensure they're development dependencies
          and aren't included as transitionary dependencies to package consumers.
          For clarity, without PrivateAssets marked here, anyone consuming Microsoft.Identity.Client
          would also be forced to install these dependencies.  PrivateAssets avoids this problem. -->
     <PackageReference Include="AsyncUsageAnalyzers" Version="1.0.0-alpha003" PrivateAssets="All" />
-    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.6.3" PrivateAssets="All" />    
+    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.6.3" PrivateAssets="All" />
```

Comment:
We generally add analyzers for all projects in the solution, i.e. tests and samples too. Good code practices need to be followed everywhere. 

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/84329e60dd751ffa0de9342bbc152e09a354017b/Directory.Build.props#L26

Reply:
Yeah those got included with VS. I will update in all the places